lib/puppet/application/lookup.rb

```diff
@@ -49,8 +57,44 @@ class Puppet::Application::Lookup < Puppet::Application
   def run_command
     options[:keys] = command_line.args
 
-    puts "Hello World!"
-    puts options
-  end
+    if options[:keys].empty?
+     raise "No keys were given to lookup."
+    end
+
+    if (options[:sort_merge_arrays] || options[:merge_hash_arrays] || options[:prefix] || options[:upack_arrays]) && options[:merge] != 'deep'
+      raise "The options #{DEEP_MERGE_OPTIONS} are only available with '--merge deep'\n#{RUNHELP}"
+    end
+
+    node = Puppet::Node.indirection.find("#{options[:node]}")
+    compiler = Puppet::Parser::Compiler.new(node)
+    compiler.compile
+    scope = compiler.topscope
+
+    use_default_value = !options[:default_value].nil?
+
+    if options[:merge]
+      if options[:merge] == 'deep'
+        merge_options = {'strategy' => 'deep',
+          'sort_merge_arrays' => !options[:sort_merge_arrays].nil?,
+          'merge_hash_arrays' => !options[:merge_hash_arrays].nil?}
 
+        if options[:prefix]
+          merge_options.merge({'prefix' => options[:prefix]})
+        end
+
+        if options[:unpack_arrays]
+          merge_options.merge({'unpack_arrays' => options[:unpack_arrays]})
+        end
+
+      else
+        merge_options = {'strategy' => options[:merge]}
+      end
+
+      value = Puppet::Pops::Lookup.lookup(scope, options[:keys], options[:type], options[:default_value], use_default_value, {}, {}, merge_options)
+    else
+      value = Puppet::Pops::Lookup.lookup(scope, options[:keys], options[:type], options[:default_value], use_default_value, {}, {}, nil)
```

Comment:
This long statement can be moved out of the containing if/else instead of being repeated twice. Just always pass merge_options but set it to nil before the if/else.